langserver/fs.go

```diff
@@ -26,42 +26,67 @@ func IsFileSystemRequest(method string) bool {
 		method == "textDocument/didSave"
 }
 
-func (h *HandlerShared) HandleFileSystemRequest(ctx context.Context, req *jsonrpc2.Request) error {
+// HandleFileSystemRequest handles textDocument/did* requests. The path the
+// request is for is returned. true is returned if a file was modified.
+func (h *HandlerShared) HandleFileSystemRequest(ctx context.Context, req *jsonrpc2.Request) (string, bool, error) {
 	span := opentracing.SpanFromContext(ctx)
 	h.Mu.Lock()
 	overlay := h.overlay
 	h.Mu.Unlock()
 
+	var before []byte
+	var beforeErr error
+	checkChanged := func(uri string) bool {
+		// We ignore readFile errors, since the []byte being nil gives
+		// us the same comparison.
```

Comment:
This comment seems out of date. Perhaps just document both the if and else cases below.